File: snobot_sim_java/src/main/java/com/snobot/simulator/simulator_components/smart_sc/SmartScLimitSwitch.java
package com.snobot.simulator.simulator_components.smart_sc;

import com.snobot.simulator.module_wrapper.ASensorWrapper;
import com.snobot.simulator.module_wrapper.interfaces.IDigitalIoWrapper;

public class SmartScLimitSwitch extends ASensorWrapper implements IDigitalIoWrapper
{
    private boolean mState;

    public SmartScLimitSwitch(int aPort)
    {
        super("CAN Limit Switch (" + (aPort - BaseCanSmartSpeedController.sCAN_SC_OFFSET) + ")");
        setWantsHidden(true);
    }

    @Override
    public boolean get()
    {
        return mState;
    }

    @Override
    public void set(boolean aState)
    {
        mState = aState;
    }

}

File: snobot_sim/src/main/native/cpp/SnobotSim/SimulatorComponents/SmartSC/SmartSCLimitSwitch.cpp
#include "SnobotSim/SimulatorComponents/SmartSC/SmartSCLimitSwitch.h"
#include "SnobotSim/SimulatorComponents/SmartSC/CanIdOffset.h"


const std::string SmartSCLimitSwitch::TYPE = "com.snobot.simulator.simulator_components.smart_sc.SmartScEncoder";

SmartSCLimitSwitch::SmartSCLimitSwitch(int aPort) :
        AModuleWrapper("CAN Limit Switch " + std::to_string(aPort - CAN_SC_OFFSET))
{

}

SmartSCLimitSwitch::~SmartSCLimitSwitch()
{

}

bool SmartSCLimitSwitch::Get()
{
    return mState;
}
void SmartSCLimitSwitch::Set(bool aState)
{
    mState = aState;
}